src/methods/clipping/intersection.jl

```diff
@@ -175,17 +190,34 @@ function _intersection_point(::Type{T}, (a1, a2)::Edge, (b1, b2)::Edge) where T
     # Second line runs from q to q + s 
     qx, qy = GI.x(b1), GI.y(b1)
     sx, sy = GI.x(b2) - qx, GI.y(b2) - qy
-    # Intersections will be where p + αr = q + βs where 0 < α, β < 1 and
+    # Intersections will be where p + αr = q + βs where 0 < α, β < 1
     r_cross_s = rx * sy - ry * sx
     Δqp_x = qx - px
     Δqp_y = qy - py
     if r_cross_s != 0  # if lines aren't parallel
-        α = (Δqp_x * sy - Δqp_y * sx) / r_cross_s
-        β = (Δqp_x * ry - Δqp_y * rx) / r_cross_s
-        x = px + α * rx
-        y = py + α * ry
+        # Calculate α (uses approx comparisons due to inexact calculations)
+        α_num_t1, α_num_t2 = Δqp_x * sy, Δqp_y * sx  # α numerator terms
+        α_num = α_num_t1 - α_num_t2
+        α = if α_num_t1 ≈ α_num_t2
+            zero(T)
+        elseif α_num ≈ r_cross_s
+            one(T)
+        else
+            T(α_num / r_cross_s)
```

Comment:
This could use a few more comments

Reply:
I will try to explain the math better. It was much cleaner before I separated it out to do the approximate comparisons. 

It's bad to do `isapprox(0, x)`, right? and I was trying to get earlier in the calculations before the machine error had the chance to spread. Is that how that works? 

Basically, I was getting a bug because `a = 0.99999999999999997` (some number of 9s) and it was supposed to be `a == 1`. I also realized the same thing could happen when checking `a == 0` so I separated out the calculations and made checks on things that would make `a == 0` or `a == 1` instead of directly comparing to 0 and 1.